docs/architecture/decisions/0005-IAsyncEnumerable-HttpResponseExceptionFilter.md

```diff
@@ -0,0 +1,24 @@
+# 5. Using IAsyncEnumerable partial failure won't be handled by HttpResponseExceptionFilter
+
+## Status
+Accepted
+
+## Context
+There are currently two endpoints (`emissions/forecasts/batch` , `emissions/average-carbon-intensity/batch`) that require passing a payload as an array of items. This array can be have as many items as the user wants, and it can take sometime to be processed which incurs in a delay to the client, which might feel that the service is irresponsive. Also, since these requests return enumerable items, and those are buffered before the client gets it, it impacts the overall memory footprint of the WebApp. By changing the signature of these endpoints to return an `IAsyncEnumerable` collection, it would help to deal with these issues.
+
+## Decision
```

Comment:
Depending on which approach you take, `Use IAsyncEnumerable: Denied` or `Use IActionResult: Accepted`, can we put the relevant pros into the  `## Decision`, and the cons into the `## Consequences` sections.